src/OpenTelemetry/Metrics/MetricReaderExt.cs

```diff
@@ -72,6 +72,10 @@ internal Metric AddMetricWithNoViews(Instrument instrument)
                     try
                     {
                         metric = new Metric(metricStreamIdentity, this.GetAggregationTemporality(metricStreamIdentity.InstrumentType), this.maxMetricPointsPerMetricStream, exemplarFilter: this.exemplarFilter);
+                        if (this.parentProvider?.GetSdkHealthReporter() != null)
+                        {
+                            metric.AddAggStoreCallback();
```

Comment:
You could probably mark `Metric.aggStore` internal and avoid adding these new methods on the Metric class.